consensus/phases/secondphase.go

```diff
@@ -301,23 +301,38 @@ func (sp *SecondPhaseImpl) Execute21(ctx context.Context, pulse *core.Pulse, sta
 	return state, nil
 }
 
-func (sp *SecondPhaseImpl) generatePhase2Bitset(list network.UnsyncList, proofs map[core.Node]*merkle.PulseProof) (packets.BitSet, error) {
+func (sp *SecondPhaseImpl) generatePhase2Bitset(list network.UnsyncList, proofs map[core.Node]*merkle.PulseProof, pulseNumber core.PulseNumber) (packets.BitSet, error) {
 	bitset, err := packets.NewBitSet(list.Length())
 	if err != nil {
 		return nil, err
 	}
 	cells := make([]packets.BitSetCell, 0)
 	for node := range proofs {
-		cells = append(cells, packets.BitSetCell{NodeID: node.ID(), State: packets.Legit})
+		cells = append(cells, packets.BitSetCell{
+			NodeID: node.ID(),
+			State:  livingNodeState(list.GetActiveNode(node.ID()), pulseNumber),
```

Comment:
`GetActiveNode` is senseless. Pass `node` instead of `list.GetActiveNode(node.ID())`